core/controller/src/main/scala/whisk/core/controller/Triggers.scala

```diff
@@ -387,15 +387,17 @@ trait WhiskTriggersApi extends WhiskCollectionAPI {
       .map(pkg => Path / pkg.namespace / rule.action.name.asString)
       .getOrElse(Path / rule.action.name.asString)
 
-    val request = HttpRequest(
-      method = POST,
-      uri = url.withPath(actionUrl ++ actionPath),
-      headers = List(
-        Authorization(BasicHttpCredentials(user.authkey.uuid.asString, user.authkey.key.asString)),
-        transid.toHeader),
-      entity = HttpEntity(MediaTypes.`application/json`, args.compactPrint))
+    user.authkey.getCredentials
+      .map { creds =>
+        val request = HttpRequest(
+          method = POST,
+          uri = url.withPath(actionUrl ++ actionPath),
+          headers = List(Authorization(creds), transid.toHeader),
+          entity = HttpEntity(MediaTypes.`application/json`, args.compactPrint))
 
-    singleRequest(request)
+        singleRequest(request)
+      }
+      .getOrElse(Future.failed(new RuntimeException("invalid credentials passed")))
```

Comment:
Need to think about which exception to throw here and which message to use. If we need to write it ourselves, let's add it to `Messages.scala`

Reply:
Since this message is not user facing I have added an custom Exception into this class